### PR TITLE
8256238: Remove Matcher::pass_original_key_for_aes

### PR DESCRIPTION
**Pull request by cl4es**
This removes Matcher::pass_original_key_for_aes() and related code. This was added to workaround some limitations with the AES provider on SPARC and is now effectively dead code.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

### Issue
 * [JDK-8256238](https://bugs.openjdk.java.net/browse/JDK-8256238): Remove Matcher::pass_original_key_for_aes

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Christian Hagedorn](https://openjdk.java.net/census#chagedorn) (@chhagedorn - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1175/head:pull/1175`
`$ git checkout pull/1175`

### CODE REVIEW

File: src/hotspot/share/opto/library_call.cpp [hunk 1]
```diff
@@ -5807,13 +5781,8 @@ bool LibraryCallKit::inline_electronicCodeBook_AESCrypt(vmIntrinsics::ID id) {
...
   if (k_start == NULL) return false;
 
-  Node* ecbCrypt;
-  if (Matcher::pass_original_key_for_aes()) {
-    // no SPARC version for AES/ECB intrinsics now.
-    return false;
-  }
   // Call the stub, passing src_start, dest_start, k_start, r_start and src_len
-  ecbCrypt = make_runtime_call(RC_LEAF | RC_NO_FP,
+  Node* ecbCrypt = make_runtime_call(RC_LEAF | RC_NO_FP,
```
Comment (TobiHartmann) [inline src/hotspot/share/opto/library_call.cpp:5785]: Indentation should be fixed.

File: src/hotspot/share/opto/library_call.cpp [hunk 2]
```diff
@@ -5891,13 +5860,8 @@ bool LibraryCallKit::inline_counterMode_AESCrypt(vmIntrinsics::ID id) {
...
   Node* used = field_address_from_object(counterMode_object, "used", "I", /*is_exact*/ false);
 
-  Node* ctrCrypt;
-  if (Matcher::pass_original_key_for_aes()) {
-    // no SPARC version for AES/CTR intrinsics now.
-    return false;
-  }
   // Call the stub, passing src_start, dest_start, k_start, r_start and src_len
-  ctrCrypt = make_runtime_call(RC_LEAF|RC_NO_FP,
+  Node* ctrCrypt = make_runtime_call(RC_LEAF|RC_NO_FP,
```
Comment (TobiHartmann) [inline src/hotspot/share/opto/library_call.cpp:5864]: Indentation should be fixed.

**Review by TobiHartmann — APPROVED** [review]
Otherwise looks good to me.

**Review by TobiHartmann — APPROVED** [review]

**Review by chhagedorn — APPROVED** [review]
Looks good!

**Comment by cl4es** [comment]
@TobiHartmann @chhagedorn - thank you for reviewing!

/integrate

**Pull request closed without merging**